libcxx/include/__configuration/abi.h

```diff
@@ -30,8 +30,20 @@
 #elif _LIBCPP_ABI_FORCE_MICROSOFT
 #  define _LIBCPP_ABI_MICROSOFT
 #else
+// Windows uses the Microsoft ABI
 #  if defined(_WIN32) && defined(_MSC_VER)
 #    define _LIBCPP_ABI_MICROSOFT
+
+// 32-bit ARM uses the ARM ABI with a few oddities (array cookies, etc),
+// and so does 64-bit ARM on Apple platforms.
+#  elif defined(__arm__) || (defined(__APPLE__) && defined(__aarch64__))
+#    define _LIBCPP_ABI_ARM_WITH_32BIT_ODDITIES
```

Comment:
@philnik777 I decided to rename this after reading https://github.com/llvm/llvm-project/blob/bdef80f7a9a1f563f7d2f528e12bb3321d836857/clang/lib/CodeGen/ItaniumCXXABI.cpp#L581-L583. Using `_LIBCPP_ABI_ARM` looked nicer, but things aren't that simple. 32-bit ARM is not the same as 64-bit ARM, and Apple 64-bit ARM is not the same as non-Apple 64-bit ARM.

Reply:
WDYT about `_LIBCPP_ABI_ITANIUM_WITH_ARM_EXCEPTIONS`?

Reply:
Exceptions as in differences, not as in `try catch`, right? In that case, what about `_LIBCPP_ABI_ITANIUM_WITH_ARM_DIFFERENCES`?

Reply:
Yeah, that's better.